File: src/utils/defoldTweens/Tween.ts
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/no-extra-non-null-assertion */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-non-null-assertion */
/* eslint-disable @typescript-eslint/no-unsafe-return */
import { TweensForNode, _newTweensForNode } from "./TweenGUI";

//TODO: 
type TweensForGO = {};

export function tween<T extends 'gui' | 'go'>(obj: string | node | hash, type: T): T extends 'go' ? TweensForGO : TweensForNode {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
    return  type == 'gui' ? _newTweensForNode(obj) : _newTweensForGO(obj) as any;
}

function _newTweensForGO(obj: hash): TweensForGO {
    throw Error('Tween for go is not implemented');
    return {

    };
}

File: app/utils/defoldTweens/Tween.lua
local ____lualib = require("lualib_bundle")
local Error = ____lualib.Error
local RangeError = ____lualib.RangeError
local ReferenceError = ____lualib.ReferenceError
local SyntaxError = ____lualib.SyntaxError
local TypeError = ____lualib.TypeError
local URIError = ____lualib.URIError
local ____exports = {}
local _newTweensForGO
local ____TweenGUI = require("utils.defoldTweens.TweenGUI")
local _newTweensForNode = ____TweenGUI._newTweensForNode
function _newTweensForGO(obj)
    error(
        Error(nil, "Tween for go is not implemented"),
        0
    )
    return {}
end
function ____exports.tween(obj, ____type)
    local ____temp_0
    if ____type == "gui" then
        ____temp_0 = _newTweensForNode(obj)
    else
        ____temp_0 = _newTweensForGO(obj)
    end
    return ____temp_0
end
return ____exports
